types/classificator/index.d.ts

```diff
@@ -0,0 +1,125 @@
+// Type definitions for classificator 0.3
+// Project: https://github.com/Wozacosta/classificator
+// Definitions by: Nixinova <https://github.com/Nixinova>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export as namespace classificator;
+
+export = Classificator;
+
+declare function Classificator(options?: Classificator.Options): Classificator.NaiveBayes;
+
+declare namespace Classificator {
+    interface ClassificationResults {
+        likelihoods: Array<{
+            category: string,
+            logLikelihood: number,
+            logProba: number,
+            proba: number,
+        }>;
+        predictedCategory: string;
+    }
+
+    interface Options {
+        /**
+         * Given an input string, tokenize it into an array of word tokens.
+         *
+         * By default, removes punctuation and splits on spaces.
+         */
+        tokenizer?: (text: string) => string[];
+
+        /**
+         * Alpha parameter of the additive smoothing operation.
+         *
+         * Defaults to `1`.
+         */
+        alpha?: number;
+
+        /**
+         * Defines how the prior probablity is calculated.
+         *
+         * If set to false, the classifier will use an uniform prior rather than a learnt one.
+         *
+         * Defaults to `true`.
+         */
+        fitPrior?: boolean;
+    }
+
+    /**
+     * Naive-Bayes Classifier
+     *
+     * This is a naive-bayes classifier that uses Laplace Smoothing.
+     *
+     * Takes an (optional) options object containing:
+     * - `tokenizer` => custom tokenization function
+     */
+    class NaiveBayes {
```

Comment:
```suggestion
    interface NaiveBayes {
```
make this export interface, otherwise in TS that is possible to use this unexported symbol:
```ts
new classificator.NaiveBayes();
```
this is not true in JS

Reply:
and the symbol could have public props (visible from outside) that are reflected in `STATE_KEYS` tuple/constant.
So it's like contract from class (interface) and state interface:
```ts
const STATE_KEYS: readonly [
    'categories',
    'docCount',
    'totalDocuments',
    'vocabulary',
    'vocabularySize',
    'wordCount',
    'wordFrequencyCount',
    'options',
];

type StateKey = typeof STATE_KEYS[number];

interface StateObject extends Record<StateKey, ...> {}

interface NaiveBayes extends StateObject  {
    ....
}
```
just an idea. For now having public propererty which is same as key in state keys should.
The input for `fromJson` though should accept this StateObject like interface IMO:

```js
module.exports.fromJson = (jsonStrOrObject) => {
```

Reply:
Resolved the first, done only the `const STATE_KEYS : readonly [...]` for the latter as that seems overly complicated.